test/unit/util/FetchUtil.test.ts

```diff
@@ -1,52 +1,76 @@
+import { PassThrough } from 'stream';
 import arrayifyStream from 'arrayify-stream';
 import { fetch } from 'cross-fetch';
 import { DataFactory } from 'n3';
+import rdfDereferencer from 'rdf-dereference';
 import { RdfToQuadConverter } from '../../../src/storage/conversion/RdfToQuadConverter';
-import { fetchDataset } from '../../../src/util/FetchUtil';
+import { fetchDataset, responseToDataset } from '../../../src/util/FetchUtil';
 const { namedNode, quad } = DataFactory;
 
 jest.mock('cross-fetch');
+jest.mock('rdf-dereference');
 
 describe('FetchUtil', (): void => {
+  const url = 'http://test.com/foo';
+
   describe('#fetchDataset', (): void => {
-    const fetchMock: jest.Mock = fetch as any;
-    const url = 'http://test.com/foo';
-    const converter = new RdfToQuadConverter();
+    const rdfDereferenceMock: jest.Mocked<typeof rdfDereferencer> = rdfDereferencer as any;
 
-    function mockFetch(body: string, status = 200): void {
-      fetchMock.mockImplementation((input: string): any => ({
-        text: (): any => body,
-        url: input,
-        status,
-        headers: { get: (): any => 'text/turtle' },
-      }));
+    function mockDereference(body: string): any {
+      rdfDereferenceMock.dereference.mockImplementation((): any => {
+        const mockStream = new PassThrough();
+        if (body.length > 0) {
+          const parts = body.slice(1, -1).split(' ').map((term): string => term.slice(1, -1));
+          mockStream.emit('data', quad(namedNode(parts[0]), namedNode(parts[1]), namedNode(parts[2])));
+        }
+        mockStream.end();
```

Comment:
Can't see the output since CI is not running yet, but my guess is things break here. You are letting your stream emit events here before a listener is attached. So nobody will see the quads. If you create a stream like this you want to use `stream.push` and not end the stream. But you can make it even easier by first creating your array of quads and then using `Readable.from`. I would even advise to have the `mockDereference` function take quads as input so you don't need to do the parsing here.

Reply:
Thanks, that helps, but the error remains: 

```bash
C:\Users\Falx\Workspaces\community-server>npm test -- test/unit/util/FetchUtil.test.ts     

> @solid/community-server@2.0.1 test
> npm run test:ts && npm run jest "test/unit/util/FetchUtil.test.ts"


> @solid/community-server@2.0.1 test:ts
> tsc -p test --noEmit


> @solid/community-server@2.0.1 jest
> jest --coverageReporters text-summary -- "test/unit/util/FetchUtil.test.ts"

 FAIL  test/unit/util/FetchUtil.test.ts                                                                                                                                                                                                                     
  ● Test suite failed to run
                                                                                                                                                                                                                                                            
    TypeError: require(...).RdfDereferencer is not a constructor

      at Object.<anonymous> (node_modules/rdf-dereference/engine-default.js:191:25)
      at Object.<anonymous> (node_modules/rdf-dereference/index.js:15:19)

                                                                                                                                                                                                                                                            
=============================== Coverage summary ===============================                                                                                                                                                                            
Statements   : Unknown% ( 0/0 )
Branches     : Unknown% ( 0/0 )
Functions    : Unknown% ( 0/0 )
Lines        : Unknown% ( 0/0 )
================================================================================
Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        3.313 s
Ran all test suites matching /test\\unit\\util\\FetchUtil.test.ts/i.
```

_Note: I'll mark this as ready to review, that might kickoff the CI?_